### ISSUE TITLE
Add more convenient ways to track/untrack items by category

### ISSUE THREAD

**Issue by SacredDuckwhale** [SacredDuckwhale/Rarity#18]
[WowAce Link](https://www.wowace.com/projects/rarity/issues/272)
> Also can you sort (mostly) pets in some way so that we can tick for isntance repetable for pets in that raid at once and not to go through long list to tick each pet separately?

3. You wish for a feature to toggle pets for tracking more easily. This makes sense, but it's also not easily done and I'd rather focus on actual bugs first before I try my hand at anything invasive like this.

> 3. Maybe toogle isnt easily done,but is grouping pets that drop from lets say firelands into shorter sub section on that long list easier which also would be helpfull .So basicly instead of having 1 big long list adding subsections to list -lets say firelands  and adding pets that drop in that raid in this subsection

Not exactly a priority right now, though I guess more control and better ways to track/untrack items (by category, level, etc.) would be useful to have.

I remember Allara saying something like this wasn't going to happen, so I'll need to find that comment and see what is the background here (maybe it's overly complicated or just not worth the time).

**Comment by SacredDuckwhale** [comment]
Allara said:

> Pets marked repeatable was very, very confusing to users. It created so many tickets over the years. It’s up to you if you want to change it back, but I think Urtgard and I agreed after years of it that it was better to not have anything set to repeatable out of the box. It’s far more confusing than having them all off.

I still think it'd be useful to have a way to toggle repeatable/tracked status by category or even zone/method etc.

See also this issue on WowAce: [Link](https://www.wowace.com/projects/rarity/issues/316)

**Comment by SacredDuckwhale** [comment]
Potentially also related: [WoWAce Link](https://www.wowace.com/projects/rarity?comment=252)

> Could you add an option to untrack every mount, toy, etc? So I can track those mounts, toys, etc that I'm only interested in easier and clearer.

**Comment by pmadison** [comment]
Hey SDW - I noticed in the FAQ section on wowace that you do not plan to address this; however, it causes a serious moment of lag when hovering over the map icon with big lists of tracked items and it's very painful to untrack every one individually ( as you have pointed out ).

Would you be interested in accepting a pull request for this? If so, could you give me a very high level idea of what you would be looking for in an implementation for this?

**Comment by SacredDuckwhale** [comment]
Hey there! If you're referring to the FAQ on WowAce, then most likely it's Allara (the creator of Rarity) who wrote that. I'm not against anything if it makes sense, but I don't have much time and so I'm mostly limited to maintenance tasks.

That's actually why there's this giant pile of issues here that I can't possibly work through on my own. Of course you're more than welcome to create a PR! In fact, it's generally a good idea to do this before making any changes just to discuss the specifics and get some feedback.

The lag is caused by the extremely slow tooltip building and/or poor coding (sort algorithms etc, not quite sure). I've looked into it before, though the code was too unorganized to focus on optimization, which I'm (very) slowly trying to change.

As for implementation details, I'd leave it up to you to come up with a design/proof of concept. There's many ways to go about it, so you'd need to narrow it down to what specific problem you're trying to solve and build a feature around that. Naturally, I can give more details once I know what you're planning to do :)

**Comment by pmadison** [comment]
Fair enough! I will think a little more about it before diving in here. I'm an industry programmer of 5 years who's been playing wow for almost 15 years, but has never contributed to an addon or open source project before. Not sure how much work I want to integrate into play just yet haha.

Does Allara still contribute to the project at all? It doesn't really matter for the purposes of this discussion, mostly just curious. It seems like you're more or less the sole maintainer from what I can see here. Trying to get a better handle on reviewers and potential for conflicting ideas.

**Comment by SacredDuckwhale** [comment]
No worries! As you can see, most issues have been pending for a long time. It's ultimately a niche project with a virtually nonexistent development team. There's no need to work on something if you don't want to! You can just do whatever you feel like, or nothing at all, since it's your own free time... obviously.

Allara hasn't played WOW since the end of Warlords of Draenor AFAIK; I've pushed all the updates since Legion and created this repository from the SVN code. Ultimately, I doubt they will ever return (and if they did, they said they probably won't work on Rarity/with git IIRC).
